images/filebeat/tests/main.tf

```diff
@@ -10,49 +11,60 @@ variable "digest" {
 
 data "oci_string" "ref" { input = var.digest }
 
-resource "random_pet" "suffix" {}
+data "imagetest_inventory" "this" {}
 
-resource "helm_release" "filebeat" {
-  name             = "elastic"
-  namespace        = "filebeat-${random_pet.suffix.id}"
-  repository       = "https://helm.elastic.co"
-  chart            = "filebeat"
-  create_namespace = true
+resource "imagetest_harness_k3s" "this" {
+  name      = "filebeat"
+  inventory = data.imagetest_inventory.this
 
-
-  values = [
-    "${file("${path.module}/values.yaml")}"
-  ]
-
-  set {
-    name  = "image"
-    value = data.oci_string.ref.registry_repo
-  }
-
-  set {
-    name  = "imageTag"
-    value = data.oci_string.ref.pseudo_tag
+  sandbox = {
+    mounts = [
+      {
+        source      = path.module
+        destination = "/tests"
+      }
+    ]
   }
 }
 
-data "oci_exec_test" "check-filebeat" {
-  digest      = var.digest
-  script      = "./check-filebeat.sh"
-  working_dir = path.module
-  depends_on  = [helm_release.filebeat]
+module "helm" {
+  source = "../../../tflib/imagetest/helm"
+  name   = "filebeat"
+  repo   = "https://helm.elastic.co"
+  chart  = "filebeat"
 
-  env = [{
-    name  = "FB_NAMESPACE"
-    value = helm_release.filebeat.namespace
-  }, ]
-}
 
+  values_files = [
+    "/tests/values.yaml",
+  ]
 
-module "helm_cleanup" {
-  depends_on = [data.oci_exec_test.check-filebeat]
-  source     = "../../../tflib/helm-cleanup"
-  name       = helm_release.filebeat.id
-  namespace  = helm_release.filebeat.namespace
+  values = {
+    image    = data.oci_string.ref.registry_repo
+    imageTag = data.oci_string.ref.pseudo_tag
+  }
 }
 
+resource "imagetest_feature" "basic" {
+  harness     = imagetest_harness_k3s.this
+  name        = "Basic"
+  description = "Basic functionality of the image."
+
+  steps = [
+    {
+      name = "Helm install"
+      cmd  = module.helm.install_cmd
+    },
+    {
+      name  = "Test filebeat"
+      cmd   = <<EOF
+            export FB_NAMESPACE=default
```

Comment:
Can we make this an environment variable in the harness perhaps?

Reply:
Good callout, honestly we don't actually have to set it anymore since the test just runs in the default namespace. I'll fix it up and remove this